types/src/receipt_merkle_proof.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{TransactionReceipt, ValidationError, H256};
+use alloy_rlp::Encodable;
+use cita_trie::MemoryDB;
+use cita_trie::{PatriciaTrie, Trie};
+use hasher::HasherKeccak;
+use std::sync::Arc;
+
+/// A Merkle proof that a transaction receipt has been included in a block.
+///
+/// Merkle proofs for transaction receipts use Ethereum's [Patricia Merkle Trie][1] data structure.
+/// The `receipt_root` field in a block is the root of the trie.
+///
+/// Requires a [`TransactionReceipt`] to generate a leaf node, and the rest of the proof proceeds
+/// from the leaf node.
+///
+/// [1]: https://ethereum.org/se/developers/docs/data-structures-and-encoding/patricia-merkle-trie/
+pub struct ReceiptMerkleProof {
+    pub proof: PatriciaTrie<MemoryDB, HasherKeccak>,
```

Comment:
I'm afraid that the original implementation of Merkle proof I gave was correct; we don't want a Merkle trie but rather we want a sequence of Merkle trie nodes (a Merkle path).